tests/test_security_path_traversal.py

```
@@ -204,3 +204,19 @@ def test_run_tailer_path_validation(tmp_path):
 
     with pytest.raises(ValueError, match="run_id"):
         tailer.tail_run("..")
+
+def test_statsdb_rebuild_path_validation(tmp_path):
+    """Test that StatsDBRebuildMixin validates run_id against path traversal."""
+    from swarm.runtime.statsdb.rebuild import StatsDBRebuildMixin
+
+    class MockStatsDB(StatsDBRebuildMixin):
+        def ingest_events(self, events, run_id):
```

Ruff (0.14.14)

[warning] Missing return type annotation for private function `ingest_events`

Add return type annotation: `int`

(ANN202)